config/initializers/doorkeeper.rb

```diff
@@ -12,6 +12,7 @@
       newflow_authenticate_user!
     else
       authenticate_user!
+      redirect_to '/signup/profile' if current_user.is_needs_profile?
```

Comment:
I don't think this is necessary/the problem because there's already this: https://github.com/openstax/accounts/blob/master/app/controllers/application_controller.rb#L8

Reply:
Oh I replied in slack instead of here... the Doorkeeper controller doesn't inherit from the ApplicationController (it inherits from ActionController::Base) so the before_action doesn't work for it.

Reply:
I was gonna ask if we can merge this, and then I started thinking that perhaps we also need to do this when the student (and educator?) feature flag is ON. Because we can expect that it is always ON from now and in the future. So this line should be moved outside and below the if/else.